src/main/java/org/apache/sling/resourceresolver/impl/ResourceResolverFactoryActivator.java

```diff
@@ -490,69 +555,26 @@ protected void deactivate() {
     private void unregisterFactory() {
         FactoryRegistration local = null;
         synchronized ( this ) {
```

Comment:
We could probably get rid of the synchronization. `factoryRegistration` should only ever get set via the `factoryRegistrationWorker` thread and is this guaranteed not to be concurrent.